auth/api/iam/metadata.go

```diff
@@ -34,3 +38,23 @@ func authorizationServerMetadata(identity url.URL) OAuthAuthorizationServerMetad
 		ClientIdSchemesSupported: clientIdSchemesSupported,
 	}
 }
+
+// clientMetadata should only be used for dids managed by the node. It assumes the provided identity URL is correct.
+func clientMetadata(identity url.URL) OAuthClientMetadata {
+	softwareID, softwareVersion, _ := strings.Cut(core.UserAgent(), "/")
+	return OAuthClientMetadata{
+		//RedirectURIs:            nil,
+		TokenEndpointAuthMethod: "client_secret_basic", // TODO: this is the default value, what do we support
```

Comment:
I'm guessing it would become `vp_token-bearer`?

Reply:
no that would require defining a client assertion using the vp_token next to the grant_type similar to RFC7523 describing a
`urn:ietf:params:oauth:client-assertion-type:jwt-bearer` and 
`urn:ietf:params:oauth:grant-type:jwt-bearer`